### DISCUSSION TITLE
How do I call OnClick event on Slider images?

### DISCUSSION THREAD

**Discussion by swaminawale** [xiaolin/react-image-gallery#712] [Q&A]
I'm using Slider only to show my contents. For all the images I am using, they all have different URL's associated and I want them to open when I click on image or when I click on description. How can I do this?

I'm using onClick in this way, but its not working. Am I doing this in wrong way? 

const images = [
  {
    original: 'https://picsum.photos/id/1015/1000/600/',
    thumbnail: 'https://picsum.photos/id/1015/250/150/',
    onclick: function(){
        window.open('www.google.com')
       }
  },
  {
    original: 'https://picsum.photos/id/1019/1000/600/',
    thumbnail: 'https://picsum.photos/id/1019/250/150/',
     onclick: function(){
       window.open('www.youtube.com')
       }
  },

**Comment by garypaduana** [comment]
This works for me and gives the image url in `e.target.src`

```
<ImageGallery items={props.items} onClick={(e) => {
    window.open(e.target.src);
}}/>
```

**Reply by mdkberry** [reply]
> This works for me and gives the image url in `e.target.src`
> 
> ```
> <ImageGallery items={props.items} onClick={(e) => {
>     window.open(e.target.src);
> }}/>
> ```

I am a bit confused where you put this code you suggested. There is already the below under the render() section
```
<ImageGallery
      items={this.images}
.../>  
```

**Reply by garypaduana** [reply]
add `onClick={(e) => { window.open(e.target.src);}}` to the `ImageGallery` tag. It does not belong in `items` objects.
If I'm not understanding your question correctly please ask it a different way.

**Reply by mdkberry** [reply]
I am thinking I have different code maybe, as my ```<ImageGallery>``` section already has ```onClick={this._onImageClick.bind(this)}```
I'll find another way around it.
